src/nprintml/net/step.py

```diff
@@ -235,16 +227,42 @@ def generate_argv(self, args, pcap_file=None, npt_file=None):
                     yield str(value)
 
         # add output path
-        outdir = self.get_output_directory(args)
+        outdir = self.get_output_directory()
         outpath = npt_file or outdir / 'netcap.npt'
         yield from ('--write_file', str(outpath))
 
-    def generate_procs(self, args, pcap_files, outdir):
+    def filter_files(self, pcap_files, outdir, labels=None):
+        skipped_files = collections.deque(maxlen=4)
+        skipped_count = 0
+
         for (pcap_file, dir_basis) in pcap_files:
-            npt_file = self.make_output_path(outdir, pcap_file, dir_basis)
+            if pcap_file:
+                pcap_output = pcap_file.relative_to(dir_basis) if dir_basis else pcap_file.name
+
+                if labels is not None and str(pcap_output) not in labels.index:
+                    skipped_files.append(pcap_output)
+                    skipped_count += 1
+                    continue
+
+                npt_file = self.make_output_path(outdir, pcap_output)
 
+                yield (pcap_file, npt_file)
+            else:
+                yield (None, None)
+
+        if skipped_count > 0 and self.args.verbosity >= 1:
+            print('Skipped', skipped_count, 'PCAP file(s) missing from labels file:')
+
+            for skipped_file in skipped_files:
+                print(f'\t{skipped_file}')
+
+            if skipped_count > len(skipped_files):
+                print('\t...')
```

Comment:
And, this is where we now check whether the pcap file has a label, before preparing it for input.